CargoBay/CargoBay.m

```diff
@@ -900,31 +900,27 @@ - (void)verifyTransactionWithMethod:(NSString *)method
         }
     }];
 
-    [manager setTaskDidReceiveAuthenticationChallengeBlock:^NSURLSessionAuthChallengeDisposition(NSURLSession * _Nonnull session, NSURLSessionTask * _Nonnull task, NSURLAuthenticationChallenge * _Nonnull challenge, NSURLCredential *__autoreleasing  _Nullable * _Nullable credential) {
+    [manager setAuthenticationChallengeHandler:^id _Nonnull(NSURLSession * _Nonnull session, NSURLSessionTask * _Nonnull task, NSURLAuthenticationChallenge * _Nonnull challenge, void (^ _Nonnull completionHandler)(NSURLSessionAuthChallengeDisposition, NSURLCredential * _Nullable)) {
         NSURLSessionAuthChallengeDisposition result = NSURLSessionAuthChallengeCancelAuthenticationChallenge;
         
         if ([[[challenge protectionSpace] authenticationMethod] isEqualToString:NSURLAuthenticationMethodServerTrust]) {
             SecTrustRef trust = [[challenge protectionSpace] serverTrust];
             NSError *error = nil;
             
-            BOOL didUseCredential = NO;
             BOOL isTrusted = CBValidateTrust(trust, &error);
             if (isTrusted) {
                 NSURLCredential *credential = [NSURLCredential credentialForTrust:trust];
                 if (credential) {
-                    result = NSURLSessionAuthChallengeUseCredential;
-                    didUseCredential = YES;
+                    return credential;
                 }
             }
             
-            if (!didUseCredential) {
-                result = NSURLSessionAuthChallengeCancelAuthenticationChallenge;
-            }
+            result = NSURLSessionAuthChallengeCancelAuthenticationChallenge;
```

Comment:
I don't think you need to set result this value again. It defaults to this value already.

Reply:
Definitely, just copy-pasted from above without noticing the same value was set before. Thanks for pointing that out